url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/dbms/model3d/TriangleElt3D.java

```diff
@@ -0,0 +1,475 @@
+				&& this.getNeighbour(1).isGeometryEquivalent(element, sop)) {
+			this.setNeighbour(1, null);
+			return 1;
+		}
+		if (this.getNeighbour(2) != null
+				&& this.getNeighbour(2).isGeometryEquivalent(element, sop)) {
+			this.setNeighbour(2, null);
+			return 2;
+		}
+
+		return -1;
+	}
+
+	/**
+	 * Returns the neighbour for the given index.
+	 * 
+	 * @param index
+	 *            int
+	 * @return the neighbouring triangle or <code>null</code> if
+	 *         <code>index</code> out of range.
+	 */
+	public TriangleElt3D getNeighbour(int index) {
+		switch (index) {
+		case 0:
+			return this.eltZero;
+		case 1:
+			return this.eltOne;
+		case 2:
+			return this.eltTwo;
+		default:
+			return null;
+		}
+	}
+
+	/**
+	 * Returns the neighbours as array (of size of neighbours count).
+	 * 
+	 * @return TriangleElt3D[] - orderd as explained.
+	 */
+	public TriangleElt3D[] getNeighbours() {
+		int counter = 0;
+		if (this.eltZero != null)
+			counter++;
+		if (this.eltOne != null)
+			counter++;
+		if (this.eltTwo != null)
+			counter++;
+
+		TriangleElt3D[] neighbours = new TriangleElt3D[counter];
+
+		counter = 0;
+		if (this.eltZero != null) {
+			neighbours[counter] = this.eltZero;
+			counter++;
+		}
+		if (this.eltOne != null) {
+			neighbours[counter] = this.eltOne;
+			counter++;
+		}
+		if (this.eltTwo != null) {
+			neighbours[counter] = this.eltTwo;
+		}
+
+		return neighbours;
+	}
+
+	/**
+	 * Returns the enclosing net.
+	 * 
+	 * @return TriangleNet3D - enclosing net.
+	 */
+	TriangleNet3D getNet() {
+		return comp.getNet();
+	}
+
+	/**
+	 * Returns the enclosing net component
+	 * 
+	 * @return TriangleNet3DComp - enclosing net component.
+	 */
+	public TriangleNet3DComp getNetComponent() {
+		return comp;
+	}
+
+	/**
+	 * Sets the reference to the enclosing net component.
+	 * 
+	 * @param net3D
+	 *            TriangleNet3DComp enclosing net component
+	 */
+	public void setNetComponent(TriangleNet3DComp net3D) {
+		comp = net3D;
+	}
+
+	/**
+	 * Sets the id to given value
+	 * 
+	 * @param i
+	 *            int to which the id should be set
+	 */
+	public void setID(int i) {
+		this.id = i;
+	}
+
+	/**
+	 * Returns the type of this as a <code>SimpleGeoObj</code>.
+	 * 
+	 * @return byte - type.
+	 * @see db3d.dbms.geom.SimpleGeoObj#getType()
+	 */
+	public byte getType() {
+		return SimpleGeoObj.TRIANGLE_ELT_3D;
+	}
+
+	/**
+	 * Checks whether neighbours of this have already been visited, makes them
+	 * orientation consistent if not.
+	 * 
+	 * @param sop
+	 *            ScalarOperator needed for validation
+	 * @param flags
+	 *            FlagMap to store the visited neighbours
+	 * @author Dag<br>
+	 *         Revision: Edgar Butwilowski
+	 * @throws IllegalArgumentException
+	 *             - if index of a triangle point is not 0, 1 or 2. The
+	 *             exception originates in the method getPoint(int) of the class
+	 *             Triangle3D.
+	 */
+	public void makeNeighboursOrientationConsistent(ScalarOperator sop,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.